test/cppsuite/tests/session_microbenchmarks.cpp

```diff
@@ -0,0 +1,230 @@
+{
+    std::string cursor_uri = tc->db.get_collection(0).name;
+
+    auto cursor = tc->session.open_scoped_cursor(cursor_uri);
+    cursor->set_key(cursor.get(), "key" + id);
+    cursor->set_value(cursor.get(), "value1" + id);
+    int ret = cursor->insert(cursor.get());
+
+    if (ret != 0) {
+        logger::log_msg(
+          LOG_ERROR, "session_microbenchmarks: expected insert id " + id + " to succeed");
+    }
+}
+
+/*
+ * Class that defines operations that do nothing as an example. This shows how database operations
+ * can be overridden and customized.
+ */
+class session_microbenchmarks : public test {
+public:
+    session_microbenchmarks(const test_args &args) : test(args)
+    {
+        init_operation_tracker(
+          new operation_tracker_session_microbenchmarks(_config->get_subconfig(OPERATION_TRACKER),
+            _config->get_bool(COMPRESSION_ENABLED), *_timestamp_manager));
+    }
+
+    void
+    run() override final
+    {
+        /* You can remove the call to the base class to fully customize your test. */
+        test::run();
+    }
+
+    void
+    background_compact_operation(thread_worker *) override final
+    {
+        logger::log_msg(LOG_WARN, "background_compact_operation: nothing done");
+    }
+
+    void
+    checkpoint_operation(thread_worker *) override final
+    {
+        logger::log_msg(LOG_WARN, "checkpoint_operation: nothing done");
+    }
+
+    void
+    custom_operation(thread_worker *tc) override final
+    {
+        WT_SESSION wt_session = *(tc->session);
+        instruction_counter begin_transaction_counter(
+          "begin_transaction_instructions", test::_args.test_name);
+        instruction_counter commit_transaction_counter(
+          "commit_transaction_instructions", test::_args.test_name);
+        instruction_counter rollback_trancation_counter(
+          "rollback_trancation_instructions", test::_args.test_name);
+        instruction_counter prepare_transaction_counter(
+          "prepare_transaction_instructions", test::_args.test_name);
+
+        instruction_counter commit_after_prepare_transaction_counter(
+          "commit_after_prepare_instructions", test::_args.test_name);
+        instruction_counter open_cursor_cached_counter(
+          "open_cursor_cached_instructions", test::_args.test_name);
+        instruction_counter open_cursor_uncached_counter(
+          "open_cursor_uncached_instructions", test::_args.test_name);
+        instruction_counter timestamp_transaction_uint_counter(
+          "timestamp_transaction_uint_instructions", test::_args.test_name);
+        std::string cursor_uri = tc->db.get_collection(0).name;
+
+        /* Get the collection to work on. */
+        testutil_assert(tc->collection_count == 1);
+
+        scoped_session &session = tc->session;
+
+        begin_transaction_counter.track([&wt_session, &session]() -> int {
+            session->begin_transaction(session.get(), NULL);
```

Comment:
I think we want to assert outside the track call that the begin returned success. Same for the other track calls.